geo-types/src/error.rs

```diff
@@ -0,0 +1,44 @@
+use std::fmt;
+
+#[derive(Debug)]
+pub enum Error {
+    MismatchedGeometry {
+        expected: &'static str,
+        found: &'static str,
+    },
+}
+
+impl std::error::Error for Error {}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::MismatchedGeometry { expected, found } => {
+                write!(f, "Expected a {}, but found a {}", expected, found)
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::{Geometry, Point, Rect};
+    use std::convert::TryFrom;
+
+    #[test]
+    fn error_output() {
+        let point = Point::new(1.0, 2.0);
+        let point_geometry = Geometry::from(point);
+
+        let rect = Rect::new(Point::new(1.0, 2.0), Point::new(3.0, 4.0));
+        let rect_geometry = Geometry::from(rect);
+
+        Point::try_from(point_geometry).expect("failed to unwrap inner enum Point");
+
+        let failure = Point::try_from(rect_geometry).unwrap_err();
+        assert_eq!(
+            format!("{}", failure),
+            "Expected a geo_types::point::Point<f64>, but found a geo_types::rect::Rect<f64>"
```

Comment:
here's what the error output actually looks like.

Reply:
An example of where I'd like to take advantage of these error methods, is in code like this:
```
 let wkt = wkt::Wkt::from_str(include_str!("../resources/test_fixtures/louisiana.wkt"))
            .expect("invalid test data");
        let ls = geo::LineString::<f32>::try_from(wkt).expect("unexpected geometry in wkt");
```

An error message like this is helpful!

> Expected a geo_types::line_string::LineString<f64>, but found a geo_types::geometry_collection::GeometryCollection<f64>

Reply:
This is great!